master/internal/api_experiment.go

```diff
@@ -235,26 +237,54 @@ func (a *apiServer) deleteExperiment(exp *model.Experiment, user *model.User) er
 	return nil
 }
 
+func protoStateDBCaseString(
+	enumToValue map[string]int32, colName, serializedName, trimFromPrefix string,
+) string {
+	query := fmt.Sprintf("CASE %s::text ", colName)
+	for enum, v := range enumToValue {
+		query += fmt.Sprintf("WHEN '%s' THEN %d ", strings.TrimPrefix(enum, trimFromPrefix), v)
+	}
+	return query + fmt.Sprintf("END AS %s", serializedName)
+}
+
 func (a *apiServer) GetExperiments(
-	_ context.Context, req *apiv1.GetExperimentsRequest,
+	ctx context.Context, req *apiv1.GetExperimentsRequest,
 ) (*apiv1.GetExperimentsResponse, error) {
-	// Construct the experiment filtering expression.
-	var allStates []string
-	for _, state := range req.States {
-		allStates = append(allStates, strings.TrimPrefix(state.String(), "STATE_"))
-	}
-	stateFilterExpr := strings.Join(allStates, ",")
-	userFilterExpr := strings.Join(req.Users, ",")
-	userIds := make([]string, 0)
-	for _, userID := range req.UserIds {
-		userIds = append(userIds, strconv.Itoa(int(userID)))
-	}
-	userIDFilterExpr := strings.Join(userIds, ",")
-	labelFilterExpr := strings.Join(req.Labels, ",")
-	archivedExpr := ""
-	if req.Archived != nil {
-		archivedExpr = strconv.FormatBool(req.Archived.Value)
-	}
+	resp := &apiv1.GetExperimentsResponse{Experiments: []*experimentv1.Experiment{}}
+	query := db.Bun().NewSelect().
+		Model(&resp.Experiments).
+		ModelTableExpr("experiments as e").
+		Column("e.id").
+		ColumnExpr("e.config->>'description' AS description").
+		ColumnExpr("e.config->>'labels' AS labels").
+		ColumnExpr("proto_time(e.start_time) AS start_time").
+		ColumnExpr("proto_time(e.end_time) AS end_time").
+		ColumnExpr(protoStateDBCaseString(experimentv1.State_value, "e.state", "state", "STATE_")).
+		Column("e.archived").
+		ColumnExpr(
+			"(SELECT COUNT(*) FROM trials t WHERE e.id = t.experiment_id) AS num_trials").
+		// Intentionally not sending trial_ids due to performance.
```

Comment:
i'm not sure how trial ids would kill us in perf but `SELECT COUNT(*) FROM trials t WHERE e.id = t.experiment_id) AS num_trials"` would not.

Reply:
The one difference I can think of is that sending trial IDs could potentially be a significant amount of IDs we need to serialize / deserialize / send over the network. 

I would be worried that it could cause someones setup to hang. Though I am honestly not sure.

Reply:
for even the largest hp searches i think this would be ~8kb max. so i guess i was thinking that the join cost would dwarf the network io. maybe not though.

Reply:
It would be cool to have some level of performance testing for situations like this (maybe expected SLAs too?).

Though I honestly have no idea how to in the "bun" ORM way to actually get a count of something without actually getting it so I added trial IDs.

Reply:
totally agree, we should add this as a tech debt item (it even came up this release, too)